src/api/selection.js

```diff
@@ -9,8 +9,21 @@ define(function () {
     // find the parent document or document fragment
     if( rootDoc.compareDocumentPosition(scribe.el) & Node.DOCUMENT_POSITION_DISCONNECTED ) {
       var currentElement = scribe.el.parentNode;
-      while(currentElement && nodeHelpers.isFragment(currentElement)) {
-        currentElement = currentElement.parentNode;
+      // Old browsers might not know about `:host` without polyfill, lets catch exception if it is the case
+      try {
+        var withinShadowRoot = currentElement.matches && currentElement.matches(':host *');
+      } catch (e) {}
+
+      // We need to use closest Shadow Root instead of Document for elements withing Shadow Root
+      // Otherwise we'll get no useful information about actual selection
```

Comment:
I've actually left the Guardian so I no longer have admin rights to the repo. This is kind of a drive-by reviewing.

I think the code should roughly be:

```
var/const/let rootCheck = isShadowDom(currentElement) ? shadowRootCheck : domRootCheck;

while(!rootCheck(currentElement)) {
  currentElement = currentElement.parentNode;
}
```

Make sense?